src/ui/handler/dblclick_zoom.js

```diff
@@ -71,15 +72,47 @@ class DoubleClickZoomHandler {
         if (!this.isEnabled()) return;
         if (e.points.length > 1) return;
 
+        const maxDelta = 30;
+
         if (!this._tapped) {
-            this._tapped = setTimeout(() => { this._tapped = null; }, 300);
+            this._tappedPoint = e.points[0];
+            this._tapped = setTimeout(() => { this._tapped = null; this._tappedPoint = null; }, 300);
         } else {
-            clearTimeout(this._tapped);
-            this._tapped = null;
-            this._zoom(e);
+            const newTap = e.points[0];
+            const firstTap = this._tappedPoint;
+
+            if (firstTap && Math.abs(firstTap.x - newTap.x) <= maxDelta && Math.abs(firstTap.y - newTap.y) <= maxDelta) {
+                e.originalEvent.preventDefault(); // prevent duplicate zoom on dblclick
+
+                const onTouchEnd = () => {
+                    if (this._tapped) { // make sure we are still within the timeout window
+                        this._zoom(e); // pass this touchstart event, as touchend events have no points
```

Comment:
Was hoping the comment would convey this but: `e` here is actually the `touchstart` event, which I'm passing through to the `touchend` handler created _within_ the `touchstart` handler here, and just ignoring the actual `touchend` event because it has no points we can use.

a) Is there a better, less-hacky solution? 
b) If not, is there a better way I can phrase this comment to describe what's going on?

Reply:
I had to re-read the comment to correct my understanding of what is happening here. What tripped me up is that `onTouchEnd` has no event parameter. 

Would `//pass the original touchstart event...` make it easier to understand? If not, I am okay with keeping the existing wording.

Reply:
:+1: thanks @asheemmamoowala, yep I think "original" makes it clearer - edited in https://github.com/mapbox/mapbox-gl-js/pull/8086/commits/61362b03c557a37164e457c518fd92bb276f56fa. I also added a comment calling attention to the fact that we're ignoring the `touchend` event. Hope that clears things up!